pages/changelog/2024-11-20-multi-modal-custom-attachments.mdx

```diff
@@ -0,0 +1,69 @@
+---
+date: 2024-11-20
+title: Full multi-modal support, including audio, images, and arbitrary files
+badge: Launch Week 2 🚀
+description: Add arbitrary attachments to LLM traces in Langfuse to observe and analyze multi-modal LLM applications.
+author: Hassieb
+---
+
+import { ChangelogHeader } from "@/components/changelog/ChangelogHeader";
+
+<ChangelogHeader />
+
+[Back in August](/changelog/2024-08-21-openai-multi-modal-traces), we added support for rendering multi-modal traces in Langfuse if they reference external files. Since then, extending multi-modal support to base64 encoded images and arbitrary attachments has been one a top requests from the community ([GitHub Discussion thread](https://github.com/orgs/langfuse/discussions/3004) with 45 upvotes).
```

Comment:
Currently we don't support arbitrary attachments, only the list of supported mimetypes. Should I prioritize extending the list? I still need any minute for the thursday cost tracking launch

Reply:
Current list:

```
  PNG = "image/png",
  JPEG = "image/jpeg",
  JPG = "image/jpg",
  WEBP = "image/webp",
  MP3 = "audio/mpeg",
  MP3_LEGACY = "audio/mp3",
  WAV = "audio/wav",
  TXT = "text/plain",
  PDF = "application/pdf",
  ```